docs/components/query.md

```diff
@@ -429,6 +429,17 @@ Flags:
                                  that are always used, even if the health check
                                  fails. Useful if you have a caching layer on
                                  top.
+      --store.grpc.samples-limit=0
+                                 The maximum samples allowed for a single
+                                 Series request, The Series call fails if
+                                 this limit is exceeded. 0 means no limit.
+                                 NOTE: For efficiency the limit is internally
+                                 implemented as 'chunks limit' considering each
+                                 chunk contains a maximum of 120 samples.
+      --store.grpc.series-limit=0
+                                 The maximum series allowed for a single Series
+                                 request. The Series call fails if this limit is
+                                 exceeded. 0 means no limit.
```

Comment:
Correct me if I'm wrong, but it seems like we have two distinct `store` flag categories:
- One is to configure the stores to which query connects
- The second one added in this PR refers to the store server of the query

I wonder if we could distinguish them more clearly. Maybe `store-limit.*`?

Reply:
Good point, should we use something like `store.limits.select-series` and `store.limits.select-samples`?

Reply:
I think that would still be mixing both in one bag, since both flag groups are still under `flag`. But on the second thought, I think we shot ourself in the foot already since even in existing code we have dual use in query vs store... 

And yet now that I'm thinking, we deprecated `store` in favor of `endpoint`, so eventually any store-connection related flags will be actually under `endpoint`. 

I guess we can reclaim `store` as config for the actual store server, then `store.limits.*` is fine 👍.

Reply:
Yeah in this case `store` would be the store server, whereas `endpoint` would be something we configure on the client. I updated the names of the flags.